src/lib/content.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { createRoot } from 'react-dom/client'
+import { mathifyElement } from './utils'
+import { ContentLoader } from '../components/ContentLoader'
+
+const OS_RAISE_CONTENT_CLASS = 'os-raise-content'
+
+export const renderContentElements = async (): Promise<void> => {
+  const osContentItems = document.querySelectorAll(`.${OS_RAISE_CONTENT_CLASS}`)
+  const contentPromises: Array<Promise<void>> = []
+
+  osContentItems.forEach((elem) => {
+    const htmlElem = elem as HTMLElement
+    const contentId = htmlElem.dataset.contentId
+
+    if (contentId === undefined) {
+      console.log('WARNING: Ignoring os-raise-content with missing UUID data attribute')
+      return
+    }
+
+    if (htmlElem.hasChildNodes()) {
+      console.log('WARNING: Found non-empty os-raise-content')
+    }
+
+    mathifyElement(elem).catch((error) => {
+      console.error(error)
+    })
+
+    const contentPromise = new Promise<void>((resolve) => {
+      createRoot(htmlElem).render(
+        <React.StrictMode>
+          <ContentLoader contentId={contentId} contentLoadedCallback={resolve}/>
```

Comment:
I'm open to alternate ways of accomplishing this, but the general problem is there are certain things (e.g. like invoking math rendering, parsing for "page types", etc.) we can only do after all of the content has been loaded and reflected in the DOM. Otherwise, for example, when we tell `MathJax` to render math it will find nothing and we'll get unrendered math when the component finally does insert the content in the DOM. This callback is invoked by `ContentLoader` when it is "done" [here](https://github.com/openstax/k12-apps-raise/blob/209a520e165b0cb5c7563dbf1ebcd2e63104c109/src/components/ContentLoader.tsx#L39-L42). The code here uses that invocation to create a Promise that we can wait on a few liens down.